### PR TITLE
Add project for running microbenchmarks

### PR DESCRIPTION
**Pull request by PaulusParssinen**
This PR adds a [BenchmarkDotNet](https://benchmarkdotnet.org/) project to the benchmarks directory for future microbenchmarks. I included some initial RESP-protocol low-level integer reading and writing benchmarks as an example, which will likely be tweaked more later. The fact that everything takes pointers instead of, say `Span<T>` makes it a little bit painful to write microbenchmarks at the moment.

Having shared benchmark suite that is precise and reproducible will allow us to substantiate future performance PRs with real data. BenchmarkDotNet also has loads of knobs to configure it for pretty much any benchmarking needs.

### CODE REVIEW

File: benchmark/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Benchmarks
+
+This directory contains projects for benchmarking gar
```
Comment (badrishc) [inline benchmark/README.md:3]: This sentence is incomplete :)

**Review by badrishc — APPROVED** [review]

**Review by TedHartMS — APPROVED** [review]

File: benchmark/README.md (hunk 1 above)
Comment (PaulusParssinen) [reply benchmark/README.md]: Oops, not sure what happened there 😄

**Comment by PaulusParssinen** [comment]
Tweaked the benchmarks a bit more and added link to the BDN best practices in the docs.

File: Garnet.sln [hunk 2]
```diff
@@ -83,6 +83,8 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ClusterStress", "playground
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Embedded.perftest", "playground\Embedded.perftest\Embedded.perftest.csproj", "{5BEDAC1F-6458-4EBA-8174-EC06B07F2132}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "BenchmarkDotNet.benchmark", "benchmark\BenchmarkDotNet.benchmark\BenchmarkDotNet.benchmark.csproj", "{9F6E4734-6341-4A9C-A7FF-636A39D8BEAD}"
+EndProject
```
Comment (TedHartMS) [inline Garnet.sln:88]: Let's rename to BDN.benchmark, so we're not repeating "benchmark"

**Review by badrishc — APPROVED** [review]

**Pull request merged**